contracts/commands/AaveStoplLossCommand.sol

```diff
@@ -0,0 +1,307 @@
+    function executeOperation(
+        address[] calldata assets,
+        uint256[] calldata amounts,
+        uint256[] calldata premiums,
+        address initiator,
+        bytes calldata params
+    ) external returns (bool) {
+        require(initiator == trustedCaller, "aapa/caller-not-initiator");
+        FlData memory flData;
+        flData.amounts = amounts;
+        flData.assets = assets;
+        flData.premiums = premiums;
+        flData.initiator = initiator;
+        flData.params = params;
+
+        console.log("X----------------------");
+        console.log("| premium", flData.premiums[0]);
+        console.log("| amount", flData.amounts[0]);
+        console.log("| payback amount", flData.amounts[0] + flData.premiums[0]);
+        console.log("X----------------------");
+
+        require(msg.sender == address(lendingPool), "aaveSl/caller-must-be-lending-pool");
+        console.log("flData.onBehalfOf", flData.onBehalfOf);
+        console.log("initiator", initiator);
+
+        // FL token / debt token
+        address debtTokenAddress = assets[0];
+        IERC20 debtToken = IERC20(debtTokenAddress);
+        console.log("debt token in command", debtToken.balanceOf(address(this)));
+        // repay debt of initiator (proxy) that equals to
+        debtToken.approve(address(lendingPool), flData.amounts[0]);
+
+        console.log("after repay");
+        (
+            address collateralATokenAddress,
+            address collateralTokenAddress,
+            address exchangeAddress,
+            address borrower,
+            EarnSwapData.SwapData memory exchangeData
+        ) = abi.decode(flData.params, (address, address, address, address, EarnSwapData.SwapData));
+
+        lendingPool.repay(debtTokenAddress, flData.amounts[0], 2, borrower);
+        require(initiator == borrower, "aaveSl/initiator-not-borrower");
+        IERC20 collateralToken = IERC20(collateralTokenAddress);
+        uint256 aTokenBalance = IERC20(collateralATokenAddress).balanceOf(borrower);
+        console.log("aTokenBalance token in command", aTokenBalance);
+        // pull tokens from proxy
+        IERC20(collateralATokenAddress).transferFrom(borrower, address(this), aTokenBalance);
+        // withdraw colateral - we use max to get all of collateral
+        lendingPool.withdraw(collateralTokenAddress, (type(uint256).max), address(this));
+        console.log(
+            "coll token in command",
+            IERC20(collateralTokenAddress).balanceOf(address(this))
+        );
+        // approve swap to be able to transfer out the collateral token - TODO change max to actual aToken/coll token  balance
+        collateralToken.approve(exchangeAddress, type(uint256).max);
+
+        uint256 paybackReceivedFromSwap = ISwap(exchangeAddress).swapTokens(exchangeData);
+        console.log("returned from swap", paybackReceivedFromSwap);
+        require(
+            paybackReceivedFromSwap > (flData.amounts[0] + flData.premiums[0]),
```

Comment:
there is no receiveAtLeast value  ? if there is then use this one 

You can also check that receiveAtLeast >= flData.amounts[0] + flData.premiums[0]

Reply:
it's only in the exchange data that we pass directly to swap contract. I think we are fine here